lib/schema_plus/indexes/active_record/connection_adapters/sqlite3_adapter.rb

```diff
@@ -0,0 +1,19 @@
+module SchemaPlus::Indexes
+  module ActiveRecord
+    module ConnectionAdapters
+      module Sqlite3Adapter
+
+        # Extends rails' remove_index to include this options:
+        #   :if_exists
+        def remove_index(table_name, *args)
```

Comment:
Seems not very DRY to have this same code repeated in abstract_adapter, postgresql_adapter and sqlite3_adapter.   I think it would be good to create something like `lib/schema_plus/indexes/remove_if_exists.rb` that defines a module `SchemaPlus::Indexes::RemoveIfExists` containing the `remove_index` method.  Then each of the adapters could just do

```
module SchemaPlus::Indexes
  module ActiveRecord
    module ConnectionAdapters
      module Sqlite3Adapter
         include RemoveIfExists
      end
    end
  end
end
```

Though if that doesn't work for some annoying metaprogramming reason, could be slightly less DRY (though arguably clearer) and instead of including the module, define a method `SchemaPlus::Indexes::RemoveIndex.if_exists` which could be used in each adapter as

```
def remove_index(table_name, *args)
   super table_name, *args if SchemaPlus::Indexes::RemoveIndex.if_exists(table_name, args)
end
```

Also, if the method in `abstract_adapter` is only applicable to mysql since the others need to override it explicitly, then maybe move it to mysql to minimize confusion?

What do you think about all that?